meta/data.go

```diff
@@ -609,12 +616,16 @@ func shardGroupDuration(d time.Duration) time.Duration {
 	return 1 * time.Hour
 }
 
-// ShardGroupInfo represents metadata about a shard group.
+// ShardGroupInfo represents metadata about a shard group. The tombstone field is important
+// because it makes it clear that a ShardGroup has been deleted, and allow the system to be
+// sure that a ShardGroup is not simply missing. If the tombstone is set, the system can safely
+// delete any associated shards.
 type ShardGroupInfo struct {
 	ID        uint64
 	StartTime time.Time
 	EndTime   time.Time
 	Shards    []ShardInfo
+	Tombstone bool
```

Comment:
Might be better make this a `time.Time` so we can later go and expunge shard groups that have been tombstoned long ago.

Also, the function names and comments refer to `Delete`ing a shard group.  I think it would be clearer to name this `DeletedAt`personally.